addons/sale_project/models/project.py

```diff
@@ -20,7 +20,7 @@ class Project(models.Model):
     sale_line_id = fields.Many2one(
         'sale.order.line', 'Sales Order Item', copy=False,
         compute="_compute_sale_line_id", store=True, readonly=False, index='btree_not_null',
-        domain="[('is_service', '=', True), ('is_expense', '=', False), ('state', '=', 'sale'), ('order_partner_id', '=?', partner_id), '|', ('company_id', '=', False), ('company_id', '=', company_id)]",
+        domain="[('is_service', '=', True), ('is_expense', '=', False), ('state', '=', 'sale'), ('order_partner_id', '=?', partner_id), ('company_id', '=?', company_id)]",
```

Comment:
Yes, the first one was a mistake. The use case we want is the following :
if the project has a company, its sale_line_id have to be in the same company
else any sale_line_id from any company is ok. Which is what the '=?' is doing. 
The first case allowed the use of sale_line_id when the company_id of the sale_line_id was set to False (which is not even possible since its a required field on sale.line)